executor/builder.go

```diff
@@ -1087,11 +1087,13 @@ func (b *executorBuilder) buildTableReader(v *plan.PhysicalTableReader) Executor
 		priority:  b.priority,
 	}
 
-	for i := range v.Schema().Columns {
-		if v.Schema().Columns[i].ID == model.ExtraHandleID {
-			break
+	if v.Schema().Len() > 0 {
```

Comment:
We can do this optimization in column prunning, so we don't need to check zero data.